src/evict/evict_lru.c

```diff
@@ -1717,6 +1717,15 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
     if (target_pages > remaining_slots)
         target_pages = remaining_slots;
 
+    /*
+     * Skip btrees other than history store if the cache pressure is high and HS content dominates
+     * cache. Evicting unclean non-HS pages can generate even more HS content and will not help with
+     * the cache pressure and will probably just amplify it further.
+     */
+    if (!WT_IS_HS(btree) && __wt_cache_almost_full(session) &&
+      __wt_cache_bytes_plus_overhead(cache, cache->bytes_hs) >= conn->cache_size / 4)
+        target_pages = (target_pages + 1) / 10;
```

Comment:
Why use this calculation? The comment says to skip the trees - but this looks like it'll just choose fewer pages generally. That's probably going to waste work as it takes relatively more scanning to choose pages when we are selecting less per tree.

OTOH: I prefer this change if it doesn't totally drown out evicting from other trees - that means we need to worry less about pathological cases when no history store content can be evicted leading to the cache being stuck full.

Reply:
Yes the comments are outdated as I actually started with completely shutting off eviction walk for non-HS pages. My intention here is indeed to make sure we at least chose some pages that do not belong to the history store.